packages/registry/src/ports/PoapRegistry/PoapRegistryProvider.ts

```diff
@@ -0,0 +1,107 @@
+import { HttpProvider } from './HttpProvider';
+import { PaginatedResult } from '../../utils/types';
+import {
+  Attribute,
+  FetchAttributeInput,
+  FetchVersionDropAttributeInput,
+  FetchAttributesInput,
+  VersionedAttribute,
+} from '../../types';
+import { RegisrtyProvider } from '../RegistryProvider';
+import { PoapGraphqlFetchProvider } from './PoapGraphqlFetchProvider';
+import {
+  Attribute_QUERY,
+  PAGINATED_Attribute_QUERY,
+  VERSION_Attribute_BY_DROP_QUERY,
+} from './queries';
+import { WhereAttribute } from './types/filter';
+import {
+  AttributeQueryResponse,
+  AttributesQueryResponse,
+  VersionedAttributeQueryResponse,
+} from './types/response';
+
+export class PoapRegistryProvider implements RegisrtyProvider {
+  private PoapGraphqlFetchProvider: PoapGraphqlFetchProvider;
+
+  constructor(apiKey: string, HttpProvider: HttpProvider) {
+    this.PoapGraphqlFetchProvider = new PoapGraphqlFetchProvider(
```

Comment:
Creating a new instance here is not pretty, would be better to accept the class directly.

Definitely not having two types of providers.